src/lib389/lib389/backend.py

```diff
@@ -1165,13 +1165,36 @@ def __init__(self, instance, be):
             'nsslapd-cachesize',
             'nsslapd-dncachememsize',
             'nsslapd-readonly',
-            'nsslapd-referral',
             'nsslapd-require-index',
+            'nsslapd-suffix'
         ]
         mt_args = [
             'orphan',
             'nsslapd-state',
+            'nsslapd-referral',
         ]
         mt = be._mts.get(be.get_suffix())
         self.add_component(be, be_args)
         self.add_component(mt, mt_args)
+
+    def get_state(self):
+        return self.get_attr_val_utf8('nsslapd-state')
+
+    def set_state(self, new_state):
+        new_state = new_state.lower()
+        suffix = self.get_attr_val_utf8('nsslapd-suffix')
+
+        if new_state not in ['backend', 'disabled',  'referral',  'referral on update']:
+            raise ValueError(f"Invalid backend state {new_state}, value must be one of the following: 'backend', 'disabled',  'referral',  'referral on update'")
+
+        # Can not change state of replicated backend
+        replicas = Replicas(self._instance)
+        try:
+            # Check if replication is enabled
+            replicas.get(suffix)
+            raise ValueError("Can not change the backend state of a replicated suffix")
```

Comment:
I am unsure to understand this restriction. If I have a consumer/hub, its mapping tree will eventually be 'referral on update'. With this test, will it be possible to set a consumer/hub mapping tree on 'referral on update' ?

Reply:
I recall it being a requirement.  Does it cause any problems?

Reply:
@mreynolds389 , if it is called for a replicated backend with mt.set_state('referral on update') then it will raise the exception "Can not change the backend state of a replicated suffix". IIRC for a consumer/hub nsslapd-state needs to be 'referral on update' will it be done ?

Reply:
I think that the restriction is fine. There is no hard requirement but the replication plugin regularly resets that state trough repl_set_mtn_state_and_referrals, so allowing an user to set the mapping tree state on a replica is very tricky:
An administrator may think he set  safely the backend in read-only mode 
  but replication may reset it to rw mode behind the scene ...

Reply:
Good point. Indeed the server itself may reset the state, mainly as referral if REPLICA_TYPE_READONLY and also during total update.
However I still have a doubt as this patch is about a CLI enhancement to allow an administrator to switch a backend/suffix into a referral (see '[Creating suffix referrals](https://access.redhat.com/documentation/en-us/red_hat_directory_server/11/html/administration_guide/configuring_directory_databases-using_referrals#Using_Referrals-Creating_Suffix_Referrals)'). The is no documented restriction regarding replication.

Reply:
Opps... my mistake. I realize I missed your point @progier389 . Setting referral/backend on a replicated suffix is useless because it will be reset. You are correct.
The procedure in admin guide should state that it does not apply on replicated suffix.
The patch is good. Ack